### DISCUSSION TITLE
Limit of 100 characters for Payee field

### DISCUSSION THREAD

**Discussion by justalittle-BIT** [nielsmaerten/ynab-buddy#42] [Ideas]
Hi!
As of now I'm used to import directly from YNAB setting a particular column of the exported file (where my bank writes a loooong description of the transaction) as "payee" field. With the CSV inport of the YNAB web app I have no problems because it automatically limit the field to the 100 max characters (valid for Payee) and in this way it also automatically categorizes the transactions as per my automated rules.
This is not working with ynab-buddy because as soon the app recognizes the very long description in the payee field it stops giving an error.

is it possible to include in the app itself an automation that limits the character imported to the 100?

Thanks!
Bruno

**Comment by nielsmaerten** [comment]
Hi Bruno,  
This sounds relatively straightforward to fix. I've created a new release here: https://github.com/nielsmaerten/ynab-buddy/releases/tag/v1.2.2-preview.1  
Can you tell me if this one works for you?

Greets,
Niels

PS: Thanks for the tip! My bank is actually doing something similar with a looong memo field. I hadn't thought of trying to use that field to auto-detect the payee, but I'll be trying that soon :)

**Reply by TsaneTips** [reply]
Hi, a somewhat late reply, but didn't want to start a new thread because it is related to this case. 

I'm using v2.0.3 and use a hook on _onTransaction_ (https://github.com/nielsmaerten/ynab-buddy/discussions/86#discussioncomment-9270926) to duplicate the memo field into the payee field (my rules are also set up to automatically categorise the transactions).
When running ynab-buddy, I get an error while trying to upload:
`Error detail: {"error":{"id":"400","name":"bad_request","detail":"payee name is too long (maximum is 100 characters) (index: 0), payee name is too long (maximum is 100 characters) (index: 1), ` etc ...

Is it possible that the payee name gets truncated before this hook is executed, and that the payee name uploaded is still over 100 characters? In the saved .csv result the memo field is duplicated into the payee field, but not limited to 100 characters.
For the record, I did succeed in uploading the transactions without using the hook and duplicated field into the payee name.

Thanks!
Tsane

**Reply by nielsmaerten** [reply]
Hi!

Yes, that's correct: Truncating the payee field happens between `onRecord` and `onTransaction`. (here, to be exact: https://github.com/nielsmaerten/ynab-buddy/blob/main/src/lib/parser.ts#L50)

So, if you want to copy the memo into the payee field, you'll have to truncate it yourself inside `onTransaction`.  
It might look like this:

```js
function onTransaction(tx, record) {
  tx.payee_name: record.memo?.trim().slice(0,99)
}
```

**Reply by TsaneTips** [reply]
Thanks, works perfect now (although I had to change the ":" to "=" for it to work)!

**Comment by justalittle-BIT** [comment]
Thanks!
That seems to be fixed
The app is now giving me another error…

> Unable to parse ''.
> The expected date format was: 'dd/MM/yyyy'.
> You may want to check the format in your config.yaml file.


But this is not applicable to the same issue for sure.

The config.yaml is set like this:

> […]
> header_rows: 10
> footer_rows: 3
> delimiter: ","
> columns: [skip, skip, date, payee, inflow, inflow, skip]
> date_format: dd/MM/yyyy
> decimal_separator: "."
> thousand_separator: ""
> […]

And the CSV is like this (added some xxxx for privacy reasons):

> […]
> ,Data contabile,Data valuta,Tipologia,Entrate,Uscite,Divisa
> ,,30/08/2022,POS-xxxxxxx,,-7.00,EUR
> ,,30/08/2022,POS-xxxxxxx,,-33.29,EUR
> ,,30/08/2022,POS-xxxxxxx,,-3.00,EUR
> ,,30/08/2022,POS-xxxxxxx,,-24.00,EUR
> ,01/09/2022,01/09/2022,Disposizione - RIF:xxxxxxx,,-180.00,EUR
> ,01/09/2022,31/08/2022,Addebito canone -xxxxxxx,,-1.00,EUR
> […]

Any clue? Should open a new issue?


Thanks again!
Bruno A. De Santis

**Comment by nielsmaerten** [comment]
Looks like ynab-buddy is selecting the wrong column here. The double empty quotes mean it's trying to parse an empty field ('') as a date.  
I don't see anything immediately wrong with your config, so I'll have to try and reproduce this to figure out what's happening.

I created a new issue here: #45 

Possibly unrelated, but 2 things do stand out to me:  
- you have 2 columns marked as inflow? does your bank put debit/credit transactions in different columns?
- it says there are 10 header rows? not impossible, just seems like a lot :) So this means the actual transactions don't start until row 11?

**Comment by justalittle-BIT** [comment]
Thanks a lot!

With reference to your doubts:
1. Yes, one column has a negative number for the outflow, if I set this as
an outflow it will read it as an income (YNAB automatically adds another
"minus" to the outflow numbers transforming it in a positive number);
2. My bank gives a summary of reconciled and non reconciled account
availability. So, a lot of rows to delete.

Ciao!

Il gio 1 set 2022, 17:30 Niels Maerten ***@***.***> ha
scritto:

**Reply by nielsmaerten** [reply]
Just a quick note: this is still on my list but I'm on vacation until next week. I'll take a closer look when I get back :)
